polars/polars-core/src/chunked_array/strings.rs

```diff
@@ -2,6 +2,36 @@ use crate::chunked_array::kernels::strings::string_lengths;
 use crate::prelude::*;
 use arrow::compute::kernels::substring::substring;
 use regex::Regex;
+#[cfg(feature = "extract_jsonpath")]
+use {
+    jsonpath_lib::Compiled,
+    serde_json::{from_str, Value},
+    std::borrow::Cow,
+    std::string::String,
+};
+
+#[cfg(feature = "extract_jsonpath")]
+fn extract_json<'a>(expr: &'a Compiled, json_str: Option<&'a str>) -> Option<Cow<'a, str>> {
+    if let Some(value) = json_str {
```

Comment:
This function can be optimized a bit. 


```rust
        let json: Value = from_str(value).unwrap();
        let result = expr.select(&json).unwrap();
        if result.is_empty() {
            None
        }
        let result = result[0]; // only unpack once
        if let Some(s) = result.as_str() {  // unpack and check in one pass
            Some(Cow::Borrowed(s))  // Don't turn into a `String` that saves a heap allocation.
        }
        else if let Some(v) = result.as_i64() {  // unpack and check in one pass
            Some(Cow::Owned(v.to_string()))  
        }
            ...  // other code



```

Reply:
For that String, initially I used Borrowed as well, but encounter this error:
```rust
 let results = expr.select(&json).unwrap();
   |                                       ----- `json` is borrowed here
...
16 |                     Some(Cow::Borrowed(s))
   |                     ^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```